File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
         ArrayList<Integer> answer = new ArrayList<Integer>();
        for (int i = 0; i < arr.length; i++) {
            if (answer.size() == 0) {
                answer.add(arr[i]);
            } else {
                if (answer.get(answer.size() - 1) == arr[i]) {
                    answer.remove(answer.size() - 1);
                } else {
                    answer.add(arr[i]);
                }
            }
        }
        if (answer.size() == 0) {
            return new int[]{-1};
        }

        int[] result = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            result[i] = answer.get(i);
        }
        return result;
    }
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    let answer = [];
    let i=0;
    while(i<arr.length){
        if(answer[answer.length-1] == arr[i]){
            answer.pop();
            i++;
        }else if(answer[answer.length-1] != arr[i]){
            answer.push(arr[i++]);
        }
    }
    
    if(answer.length == 0) return [-1];
    return answer;
}